### PR TITLE
Skip private accessors

### LINKED ISSUE #332: Should skip over package-private accessor in case class
For instance, in the standard library,
```scala
final case class :: [+A](override val head: A, private[scala] var next: List[A @uncheckedVariance])
```

The method `$colon$colon.next$access$1()` is the getter of field `next` and it should be skipped.

## How to fix

In `ScalaStepFilter` we can skip methods matching `$access$\d+`.

Can the field also be lazy? What should we do?

### PR DESCRIPTION
**Pull request by iusildra**
Fixes #332 
Only works for Scala 2 as there are no methods named `xxx$access$\d+` in Scala 3
Also remvoved `final` attribute from `FakeJdiMethod` and `FakeJdiType` to be able to create anon classes to solve `NotImplementedError` at call site

### CODE REVIEW

**Comment by iusildra** [comment]
I should move the check `isPackagePrivate` check to `Scala2Unpickler`, wdyt ?

**Comment by adpi2** [comment]
> I should move the isPackagePrivate check to Scala2Unpickler, wdyt ?

I think it is fine to keep it in `ScalaUnpickler` for now.

File: modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/ScalaUnpickler.scala [hunk 1]
```diff
@@ -143,6 +144,9 @@ abstract class ScalaUnpickler(scalaVersion: ScalaVersion, testMode: Boolean) ext
 
   private def skipTraitInitializer(method: Method): Boolean =
     method.bytecodes.toSeq == Seq(ByteCodes.RETURN)
+
+  private def isPrivateAccessor(method: Method): Boolean =
+    method.name.matches(""".+\$access\$\d+""")
```
Comment (iusildra) [inline modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stacktrace/ScalaUnpickler.scala:149]: I thought it did not work in Scala 2.12

**Review by iusildra — COMMENTED** [review]
Looks good, but I can't approve 😆

**Pull request merged**